bitrise.yml

```diff
@@ -1,64 +1,80 @@
 ---
-format_version: "11"
+format_version: '11'
 default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
 project_type: flutter
 
-workflows:
-
+pipelines:
   primary:
-    description: |
-      Builds project and runs tests.
+    stages:
+    - analyze-and-test-stage: { }
+    - export-stage: { }
+stages:
+  analyze-and-test-stage:
+    abort_on_fail: true
+    workflows:
+    - analyze: { }
+    - test: { }
+  export-stage:
+    abort_on_fail: true
+    workflows:
+    - export-ios-app: { }
+    - export-android-apk: { }
 
-      Next steps:
-      - Check out [Getting started with Flutter apps](https://devcenter.bitrise.io/en/getting-started/getting-started-with-flutter-apps.html).
+workflows:
+  _setup:
     steps:
-    - git-clone@6: { }
-    - flutter-installer@0: { }
-    - cache-pull@2: { }
-    - flutter-test@1:
+    - git-clone: { }
+    - restore-dart-cache: { }
+    - flutter-installer:
         inputs:
-        - project_location: $BITRISE_FLUTTER_PROJECT_LOCATION
-    - cache-push@2: { }
-    - deploy-to-bitrise-io@2: { }
-
-  deploy:
+        - is_update: false
+  _teardown:
+    steps:
+    - save-dart-cache: { }
```

Comment:
I'm not sure it's a good idea to save the cache in every stage's every workflow. For example, if I make a mistake that passes the analyze and test workflows, but fails building the iOS/Android app, the cache probably shouldn't be saved.

Reply:
Moved this to the Android export.